experiments/models.py

```diff
@@ -177,29 +178,41 @@ class Meta:
     # Encrypted field for extra security
     data = EncryptedTextField(
         _("experiments:models:datapoint:data"),
+        null=True
     )
 
+    # For binary uploads
+    file = FileField(null=True)
+
     date_added = models.DateTimeField(
         _("experiments:models:datapoint:date_added"),
         auto_now_add=True
     )
 
-    def __str__(self):
-        return "Datapoint {}".format(self.number)
-
     session = models.ForeignKey(
         'ParticipantSession', on_delete=models.CASCADE, null=True)
 
     STATUS_TEST = _('experiments:models:datapoint:label:test')
     STATUS_PILOT = _('experiments:models:datapoint:label:pilot')
 
+    def is_file(self):
+        return self.file is not None
+
     def get_status_display(self):
         if self.session is None:
             return self.STATUS_TEST
         if self.session.experiment_state == Experiment.PILOTING:
             return self.STATUS_PILOT
         return self.STATUS_TEST
 
+    def save(self, *args, **kwargs):
+        if self.data is None and self.file is None:
+            raise ValueError('Datapoint must contain either inline data or a file upload')
```

Comment:
If both self.data is None and self.file is None is a problem indeed, the data point has no Identity. However, if both self.data and self.file are not None, does it have a double identity/split personality?
Although I do understand this check is probably ment to avoid to save a datapoint that is empty.